wrangler-core/src/main/java/io/cdap/wrangler/executor/RecipePipelineExecutor.java

```diff
@@ -164,4 +155,15 @@ public void destroy() {
   public List<ErrorRecord> errors() {
     return collector.get();
   }
+
+  private List<Directive> getDirectives() throws RecipeException {
+    if (directives == null) {
+      try {
+        this.directives = recipeParser.parse();
+      } catch (DirectiveParseException | DirectiveNotFoundException | DirectiveLoadException e) {
+        throw new RecipeException(e.getMessage(), e);
```

Comment:
There are a lot of places like this in Wrangler. For this particular case, I think it wants the message of the exception to not carrying the exception class name (if we do `new RecipeException(e)`, the default logic in `Throwable` is to use the `e.toString()` as the message, which the default implementation would insert the class name there). And the reason for that is probably because the message is going to be shown in the UI.